### PR TITLE
Implement delete function for profile

### PR DESCRIPTION
**Pull request by weidak**
# Pull Request

Created delete functionality for profile

## Description

- Created button to delete
- Created a confirmation modal to confirm deletion

## Related Issue(s)

#68 

## Changes Made

Updated UI for edit profile, implemented delete user functionality

## Screenshots (if applicable)

Delete button:
image

Confirmation modal:
image

## Checklist
- [x] I have checked that the changes included in the PR are intended to merge to `master` or any destination branch.
- [x] I have verified that the new changes do not break any existing functionalities, unless the new changes are intended and have approved by the team.
- [x] I will take care of the merging and delete the side-branch after the PR is merged.

### CODE REVIEW

**Review by tryyang2001 — APPROVED** [review]
Neat and clean! LGTM.

**Comment by weidak** [comment]
I will merge this after merging ly's PR, thanks @tryyang2001!

File: frontend/src/components/profile/DeleteModal.tsx
Comment (tlyi) [inline frontend/src/components/profile/DeleteModal.tsx:24]: Could you call logOut() from AuthContext instead? This will handle removing the userId from session storage and the re-routing.

Comment (weidak) [reply frontend/src/components/profile/DeleteModal.tsx]: Ok! thanks for catching this

**Pull request merged**
